.evg.yml

```diff
@@ -36,12 +36,14 @@ buildvariants:
       - name: "build"
       - name: "test-unit"
       - name: "test-integration"
+
   - name: release
     display_name: "Release"
     expansions:
       _platform: ubuntu1804-64-jdk-8
     run_on: ubuntu1804-build
     tasks:
+      - name: "test-smoke"
       - name: "publish-maven"
```

Comment:
I looked at the logs from the Evergreen run and publish-maven is rebuilding the jars.
 [2022/04/06 16:21:58.327] Tasks to be executed: [task ':compileJava', task ':processResources', task ':classes', task ':jar', task ':generateMetadataFileForMavenJavaPublication', task ':generatePomFileForMavenJavaPublication', task ':initializeSonatypeStagingRepository', task ':javadoc', task ':javadocJar', task ':shadowJar', task ':sourceJar', task ':signMavenJavaPublication', task ':publishMavenJavaPublicationToSonatypeRepository', task ':publishToSonatype', task ':publishMaven', task ':demo:publishMaven', task ':smoketest:publishMaven']
 
 I also see that it is trying to run "smoketest:publishMaven".
 I am wondering if we could have the Gradle handle this all the way through and not have to run a special task from Evergreen?
 Because right now, the jar we smoke test and the one we publish are not the same ones. They are built the same way and I see little possibility of mishap, but still, I would prefer to smoke test the "final" artifacts.
 
 Maybe in deploy.gradle, we could have for both snapshot and releases a dependency on smoketest?
 Or I am thinking if the jars files are "ready" when the publish task runs that it will not rebuild them?
 Right now, the logs show something hinting that it's not the case :
 ` [2022/04/06 16:22:34.528] Build cache key for task ':shadowJar' is b641979972b8ba0cf1a9212a3614588c

 [2022/04/06 16:22:34.528] Task ':shadowJar' is not up-to-date because:

 [2022/04/06 16:22:34.528]   No history is available.

 [2022/04/06 16:22:34.528] *******************`

Reply:
Not quite sure why it runs the smoketest:publishMaven task.  But yes, it's a different jar file since they are in different tasks.  I was thinking as they are built in the same manner with the shadowJar that it would still be valid.  I will see what can be done in gradle to test the same jar file.

Reply:
Not sure why either, it seems that it runs all the project tasks.
Yes, the jars are built the same way, I think it just modifies or add the pom file, and sign the jar.
So, the chances of a problem are very, very low. If you don't find a way to run smoke test from maven publish or cache the jars which have been smoke tested easily, I'd say, don't worry and we can keep it the way it is done right now.

Reply:
Created investigation ticket to look into whether the smoke test can be run on the same jars that are published: [SQL-852](https://jira.mongodb.org/browse/SQL-852)